tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge_wrapper/signup_info_singleton.cpp

```diff
@@ -0,0 +1,102 @@
+SignupInfo* deserialize_signup_info(
+    const std::string&  serialized_signup_info) {
+    SignupInfo* signup_info = new SignupInfoSingleton();
+    tcf_err_t result = signup_info->DeserializeSignupInfo(
+        serialized_signup_info);
+
+    return signup_info;
+}  // deserialize_signup_info
+
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+std::map<std::string, std::string> SignupInfoSingleton::CreateEnclaveData() {
+    tcf_err_t presult;
+    // Create some buffers for receiving the output parameters
+    // CreateEnclaveData will resize appropriately
+    StringArray public_enclave_data(0);
+    Base64EncodedString sealed_enclave_data;
+    Base64EncodedString enclave_quote;
+
+    SignupDataSingleton signup_data;
+    // Create the signup data
+    presult = signup_data.CreateEnclaveData(
+        public_enclave_data, sealed_enclave_data, enclave_quote);
+    ThrowTCFError(presult);
+
+    // Parse the json and save the verifying and encryption keys
+    std::string verifying_key;
+    std::string encryption_key;
+    std::string encryption_key_signature;
+
+    SignupInfo signup_info;
+    presult = signup_info.DeserializePublicEnclaveData(
+        public_enclave_data.str());
+    ThrowTCFError(presult);
+
+    // Save the information
+    std::map<std::string, std::string> result;
+    result["verifying_key"] = signup_info.verifying_key;
+    result["encryption_key"] = signup_info.encryption_key;
+    result["encryption_key_signature"] = signup_info.encryption_key_signature;
+    result["sealed_enclave_data"] = sealed_enclave_data;
+    result["enclave_quote"] = enclave_quote;
+
+    return result;
+}  // SignupInfoSingleton::CreateEnclaveData
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+std::map<std::string, std::string> SignupInfoSingleton::UnsealEnclaveData() {
+    tcf_err_t presult;
+    // UnsealEnclaveData will resize appropriately
+    StringArray public_enclave_data(0);
+
+    SignupDataSingleton signup_data;
+    presult = signup_data.UnsealEnclaveData(
+        public_enclave_data);
+    ThrowTCFError(presult);
+
+    SignupInfo signup_info;
+    presult = signup_info.DeserializePublicEnclaveData(
+        public_enclave_data.str());
+    ThrowTCFError(presult);
+
+    std::map<std::string, std::string> result;
+    result["verifying_key"] = signup_info.verifying_key;
+    result["encryption_key"] = signup_info.encryption_key;
+    result["encryption_key_signature"] = signup_info.encryption_key_signature;
+
+    return result;
+}  // SignupInfoSingleton::UnsealEnclaveData
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+size_t SignupInfoSingleton::VerifyEnclaveInfo(const std::string& enclave_info,
+    const std::string& mr_enclave) {
+
+    SignupDataSingleton signup_data;
+    tcf_err_t result = signup_data.VerifyEnclaveInfo(
+        enclave_info, mr_enclave);
+    size_t verify_status = result;
+    return verify_status;
```

Comment:
Remove either result of verify_status variable. Can we cast type as `size_`t in return statement `return (size_t)verify_status`

Reply:
Done